src/compontents/BurgerConstructor/BurgerConstructor.jsx

```diff
@@ -75,8 +73,9 @@ const BurgerConstructor = () => {
                     :
                     <ul className={BurgerConstructorStyles.list}>
                         {cartState.cartIngredients.map((item, idx) => {
+                            const generateId = v4()
                             return <IngredientConstructorItem
-                                key={item._id + idx}
+                                key={generateId}
```

Comment:
нужно исправить: если создавать id тут, то он будет создаваться заново на каждый перерендер BurgerConstructor, и реакт будет перерендеривать все элементы списка, даже те которые не менялись

создавать этот id нужно вот тут https://github.com/Kettel1/react-burger/blob/sprint-2/step-2/src/compontents/BurgerConstructor/BurgerConstructor.jsx#L27, чтобы он создавался только один раз при добавлении ингредиента в конструктор

```js
            } else if (item.type !== 'bun' && cartState.cartBun.length !== 0) {
                item.uid = v4()
                dispatch({type: ADD_INGREDIENTS_TO_CART, ingredients: item})
            }
```

и потом уже этот uid можно будет использовать в качестве key в рендере этого списка

Reply:
Если применять такой метод, то выходит ошибка что одинаковые ключи у детей, я не понимаю как он формирует два одинаковых ключа, и почему это происходит вообще: 
Warning: Encountered two children with the same key, `4aa4dba6-8a73-4c83-9a06-fcbff71d126d`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.

Reply:
Реализовал не много инчае